From 696060ba5e89da4d60a7ee1187c8cfd298d50031 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Mon, 5 Aug 2024 10:08:34 +0100
Subject: [PATCH] docs: Add a note about removal of VMware Tools on Windows

We use VMware's recommended method for removal, but we know from
experience that this rarely works.  The alternate method being
proposed involves making very invasive changes to the Registry and
filesystem, which are inappropriate for virt-v2v to do and highly
risky.  Therefore simply document this, with notes for the virt-v2v
user if they want to try the risky method.

Fixes: https://issues.redhat.com/browse/RHEL-51169
Thanks: Yan Vugenfirer
Reported-by: Ming Xie
(cherry picked from commit c952f310c902e438a8b0b5240a4b486b698bede8)
---
 docs/virt-v2v.pod | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/docs/virt-v2v.pod b/docs/virt-v2v.pod
index c173563b..6df9d88b 100644
--- a/docs/virt-v2v.pod
+++ b/docs/virt-v2v.pod
@@ -1248,6 +1248,21 @@ B<N.B.> Take care not to interrupt the automatic driver installation
 process when logging in to the guest for the first time, as this may
 prevent the guest from subsequently booting correctly.
 
+=head3 Removing VMware Tools from Windows guests
+
+Virt-v2v attempts to remove VMware Tools.  For Windows guests this is
+supposed to happen during the first boot after conversion.
+
+We use VMware's recommended uninstallation method as that is the
+safest choice.  However our experience has shown that this method
+usually fails.  If so, VMware Tools must be removed by some other
+method.
+
+One possible method is described here:
+L<https://gist.github.com/broestls/f872872a00acee2fca02017160840624>
+You should carefully check this script since it makes very invasive
+changes to the Windows Registry and filesystem.
+
 =head2 Free space for conversion
 
 =head3 Free space in the guest
